Dockerfile

```diff
@@ -1,42 +1,49 @@
-FROM golang:1.19-alpine AS builder
+#syntax=docker/dockerfile-upstream:1.4
+FROM golang AS builder
 
 LABEL maintainer "Derek Collison <derek@nats.io>"
 LABEL maintainer "Waldemar Quevedo <wally@nats.io>"
 
-WORKDIR $GOPATH/src/github.com/nats-io/
+ARG TARGETARCH
 
-RUN apk add -U --no-cache git binutils
+ARG VERSION_NATS
+ARG VERSION_NATS_TOP
+ARG VERSION_NSC
+ARG VERSION_STAN
 
-RUN go install github.com/nats-io/nats-top@v0.5.3
+ENV GOPATH /go/${TARGETARCH}
 
-RUN go install -ldflags="-X main.version=2.7.6" github.com/nats-io/nsc/v2@v2.7.6
+RUN <<EOT 
+    mkdir -p ${GOPATH}
 
-RUN go install github.com/nats-io/natscli/nats@v0.0.35
+    go install -ldflags="-X main.version=${VERSION_NSC}" github.com/nats-io/nsc/v2@v${VERSION_NSC}
+    go install github.com/nats-io/nats-top@v${VERSION_NATS_TOP}
+    go install github.com/nats-io/natscli/nats@v${VERSION_NATS}
+    go install github.com/nats-io/stan.go/examples/stan-pub@v${VERSION_STAN}
+    go install github.com/nats-io/stan.go/examples/stan-sub@v${VERSION_STAN}
+    go install github.com/nats-io/stan.go/examples/stan-bench@v${VERSION_STAN}
+EOT
 
-RUN go install github.com/nats-io/stan.go/examples/stan-pub@latest
-RUN go install github.com/nats-io/stan.go/examples/stan-sub@latest
-RUN go install github.com/nats-io/stan.go/examples/stan-bench@latest
+FROM base
 
-FROM alpine:3.17.0
+ARG TARGETARCH
 
-RUN apk add -U --no-cache ca-certificates figlet jq
+COPY --from=builder /go/${TARGETARCH}/bin/* /usr/local/bin
 
-COPY --from=builder /go/bin/* /usr/local/bin/
+RUN <<EOT
+    addgroup -g 1000 nats
```

Comment:
I got the following when running `docker buildx bake` on the M1, not sure if environment related...

```
#0 0.041 container_linux.go:380: starting container process caused: exec: "/bin/sh": stat /bin/sh: no such file or directory  
------                                                                                                                        
Dockerfile:33
--------------------
  32 |     
  33 | >>> RUN <<EOT
  34 | >>>     addgroup -g 1000 nats
  35 | >>>     adduser -D -u 1000 -G nats nats
  36 | >>>     apk add -U --no-cache ca-certificates figlet jq
  37 | >>> EOT
  38 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c     addgroup -g 1000 nats\n    adduser -D -u 1000 -G nats nats\n    apk add -U --no-cache ca-certificates figlet jq\n" did not complete successfully: exit code: 1
```

Reply:
I'm also on an M1 system and I don't have the issue. That stage runs in an alpine image and the error is saying it can't find/execute `/bin/sh` which is odd.

Are you are able to execute `docker run --rm -it alpine:3.17.1` and receive a normal shell?

Reply:
ok got it to work, yes it was an env thing... sorry for the confusion